lib/eslint.js

```diff
@@ -476,27 +476,28 @@ module.exports = (function() {
     /**
      * Parses text into an AST. Moved out here because the try-catch prevents
      * optimization of functions, so it's best to keep the try-catch as isolated
-     * as possible
+     * as possible.
+     * This is equivalent to calling parser.parse with some exceptions:
+     *  - Errors are returned rather than thrown
+     *  - Empty input returns null
+     *  - Shebangs are ignored
      * @param {string} text The text to parse.
      * @param {Object} config The ESLint configuration object.
-     * @returns {ASTNode} The AST if successful or null if not.
+     * @returns {ASTNode|Error} The AST if successful or the error object if not.
      * @private
      */
-    function parse(text, config) {
+    api.parse = function(text, config) {
```

Comment:
Sort of, the two issues are very closely related. In the context of #1025 I mostly wanted a way to get a standard AST for the test without  having to reimplement the entirety of parse in the test, hence the private flag remaining. 
That is basically the same requirement as #948, so technically it fixes that too, but in that context it would need to be a properly tested and documented method, and I'd rather not put the effort in there until I know we have an agreed approach.